ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -439,108 +784,101 @@ def _useRealization(self, info, rlz):
-    # for more details refer to GRP-Raven-development/Disceret_opt channel on MS Teams
-
-    # 5 @ n-1: Survivor Selection from previous iteration (children+parents merging from previous generation)
-
-    # 5.1 @ n-1: fitnessCalculation(rlz)
-    # perform fitness calculation for newly obtained children (rlz)
-
-    offSprings = datasetToDataArray(rlz, list(self.toBeSampled))
-    objectiveVal = list(np.atleast_1d(rlz[self._objectiveVar].data))
-
-    # collect parameters that the constraints functions need (neglecting the default params such as inputs and objective functions)
-    constraintData = {}
-    if self._constraintFunctions or self._impConstraintFunctions:
-      params = []
-      for y in (self._constraintFunctions + self._impConstraintFunctions):
-        params += y.parameterNames()
-      for p in list(set(params) -set([self._objectiveVar]) -set(list(self.toBeSampled.keys()))):
-        constraintData[p] = list(np.atleast_1d(rlz[p].data))
-    # Compute constraint function g_j(x) for all constraints (j = 1 .. J)
-    # and all x's (individuals) in the population
-    g0 = np.zeros((np.shape(offSprings)[0],len(self._constraintFunctions)+len(self._impConstraintFunctions)))
-
-    g = xr.DataArray(g0,
-                     dims=['chromosome','Constraint'],
-                     coords={'chromosome':np.arange(np.shape(offSprings)[0]),
-                             'Constraint':[y.name for y in (self._constraintFunctions + self._impConstraintFunctions)]})
-    # FIXME The constraint handling is following the structure of the RavenSampled.py,
-    #        there are many utility functions that can be simplified and/or merged together
-    #        _check, _handle, and _apply, for explicit and implicit constraints.
-    #        This can be simplified in the near future in GradientDescent, SimulatedAnnealing, and here in GA
-    for index,individual in enumerate(offSprings):
-      newOpt = individual
-      opt = {self._objectiveVar:objectiveVal[index]}
-      for p, v in constraintData.items():
-        opt[p] = v[index]
-
-      for constIndex, constraint in enumerate(self._constraintFunctions + self._impConstraintFunctions):
-        if constraint in self._constraintFunctions:
-          g.data[index, constIndex] = self._handleExplicitConstraints(newOpt, constraint)
-        else:
-          g.data[index, constIndex] = self._handleImplicitConstraints(newOpt, opt, constraint)
-    offSpringFitness = self._fitnessInstance(rlz,
+    population = datasetToDataArray(rlz, list(self.toBeSampled))
+
+    # Handle objective values differently for single and multi-objective cases
+    if self._isMultiObjective:
+        objectiveVal = []
+        for i in range(len(self._objectiveVar)):
+            objectiveVal.append(list(np.atleast_1d(rlz[self._objectiveVar[i]].data)))
+    else:
+        objectiveVal = list(np.atleast_1d(rlz[self._objectiveVar[0]].data))
```

Comment:
Why do we need to handle single and multi-objective cases differently here? Can we define a list of list for 'objectiveVal'? I think we should avoid to use different data structures here since single objective is specific case for multi-objective. If the codes works for multi-objective, it should also work for single objective.

Reply:
Fixed in 1ca5375f74183c0de72841410aeee571a0b8fa88